src/Stratis.Bitcoin/Utilities/Guard.cs

```diff
@@ -46,6 +48,23 @@ public static T NotNull<T>(T value, string parameterName)
             return value;
         }
 
+        /// <summary>
+        /// Checks an object is not null.
+        /// </summary>
+        /// <param name="value">The object.</param>
+        /// <exception cref="ArgumentNullException">An exception if the object passed is null.</exception>
+        public static T NotNull<T>(T value)
+        {
+            // throw if the value is null
+            if (ReferenceEquals(value, null))
```

Comment:
did you do this just incase == is overloaded?  ie rather than `value == null` ?

Reply:
To keep it consistent with the other overload that uses `ReferenceEquals`

Reply:
Yep, exactly that @majikandy . We have many overloaded == operators in the code and we can't guarantee that they handle null correctly.